dotCMS/src/main/java/com/dotcms/api/APIProvider.java

```diff
@@ -0,0 +1,155 @@
+ *                 .withContentHelper(contentHelper).build();}
+
+ *
+ */
+public class APIProvider {
+
+    public final IdentifierAPI identifierAPI;
+    public final HostAPI hostAPI;
+    public final LanguageAPI languageAPI;
+    public final FileAssetAPI fileAssetAPI;
+    public final VersionableAPI versionableAPI;
+    public final UserAPI userAPI;
+    public final ContentletAPI contentletAPI;
+    public final HTMLPageAssetAPI htmlPageAssetAPI;
+    public final CategoryAPI categoryAPI;
+    public final ContentHelper contentHelper;
+    public final PermissionAPI permissionAPI;
+
+    /**
+     * main constructor
+     * @param identifierAPI
+     * @param hostAPI
+     * @param languageAPI
+     * @param fileAssetAPI
+     * @param versionableAPI
+     * @param userAPI
+     * @param contentletAPI
+     * @param htmlPageAssetAPI
+     * @param categoryAPI
+     * @param contentHelper
+     */
+
+    private APIProvider(final IdentifierAPI identifierAPI,
+            final HostAPI hostAPI,
+            final LanguageAPI languageAPI,
+            final FileAssetAPI fileAssetAPI, final VersionableAPI versionableAPI,
+            final UserAPI userAPI,  final ContentletAPI contentletAPI,
+            final HTMLPageAssetAPI htmlPageAssetAPI,
+            final CategoryAPI categoryAPI,
+            final ContentHelper contentHelper,
+            final PermissionAPI permissionAPI) {
+        this.identifierAPI = identifierAPI;
+        this.hostAPI = hostAPI;
+        this.languageAPI = languageAPI;
+        this.fileAssetAPI = fileAssetAPI;
+        this.versionableAPI = versionableAPI;
+        this.userAPI = userAPI;
+        this.contentletAPI = contentletAPI;
+        this.htmlPageAssetAPI = htmlPageAssetAPI;
+        this.categoryAPI = categoryAPI;
+        this.contentHelper = contentHelper;
+        this.permissionAPI = permissionAPI;
+    }
+
+    public static class Builder {
+
+        private IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
+        private HostAPI hostAPI = APILocator.getHostAPI();
+        private LanguageAPI languageAPI = APILocator.getLanguageAPI();
+        private FileAssetAPI fileAssetAPI = APILocator.getFileAssetAPI();
+        private VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+        private UserAPI userAPI = APILocator.getUserAPI();
+        private ContentletAPI contentletAPI = APILocator.getContentletAPI();
+        private HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+        private CategoryAPI categoryAPI = APILocator.getCategoryAPI();
+        private ContentHelper contentHelper = ContentHelper.getInstance();
+        private PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+        public Builder withIdentifierAPI(IdentifierAPI identifierAPI) {
+            this.identifierAPI = identifierAPI;
+            return this;
+        }
+
+        public Builder withHostAPI(HostAPI hostAPI) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'hostAPI' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5685344)